### ISSUE TITLE
Hacks + Worldguard + skin change = item dupe

### ISSUE THREAD

**Issue by s1h4d0w** [SkinsRestorer/SkinsRestorerX#468]
## Describe the bug
Players on my anarchy server can use speed and scaffold hacks in a protected worldguard region, do `/skin <name>` and whatever they used to scaffold with will turn into a stack of items

## How can we reproduce what you got?
1. Turn on scaffold hack
2. Turn on speed hack
3. Jump and run while scaffolding in a protected worldguard region where player is not allowed to break/place
4. Use `/skin <name>`
5. Whatever you used to scaffold will now be a full stack

**Expected behavior**
No item duplication

## Server information
Paper 1.12.2

**What version? Do NOT answer LATEST**
* Working UUID API count: 3
* Working Profile API count: 3
* The plugin currently is in a working state.
* SkinsRestorer v14.0.2
* Server: git-Paper-1618 (MC: 1.12.2)
* BungeeMode: false

## Logs
https://pastebin.com/jTxY0eWb

**Comment by xknat** [comment]
Hey there!
Thank you for taking your time to report this to us.
Sadly, i think this might be something that we can not improve / fix.
Do you know if this is only with skinsrestorer or any `entity.CraftPlayer -> refreshPlayer` (see our [PaperSkinRefresher.java](https://github.com/SkinsRestorer/SkinsRestorerX/blob/master/src/main/java/net/skinsrestorer/bukkit/skinfactory/PaperSkinRefresher.java))

**Comment by s1h4d0w** [comment]
Honestly no idea, I understand what you’re telling me but I’m not a plugin dev and sadly don’t know java. Do you know of any other commands in general plugins such as essentials, worldedit, worldguard etc that trigger a refreshPlayer? Then I could have one of my players test if that works too (and take appropriate action with those plugins too).

**Comment by xknat** [comment]
Lib's Disguises, changeskin

**Comment by xknat** [comment]
@s1h4d0w any update on this?
I think it goes out of our scope.
I see your using paper, this means that your using [PaperSkinRefresher.java ](https://github.com/SkinsRestorer/SkinsRestorerX/blob/9ee1107a756fca571a898b89bf7e871e4ad17fef/src/main/java/net/skinsrestorer/bukkit/skinfactory/PaperSkinRefresher.java)
In there we use the following :
```
            MH_REFRESH = lookup.findVirtual(ReflectionUtil.getBukkitClass("entity.CraftPlayer"), "refreshPlayer", MethodType.methodType(Void.TYPE));
            MH_GET_HANDLE = lookup.findVirtual(ReflectionUtil.getBukkitClass("entity.CraftPlayer"), "getHandle", MethodType.methodType(ReflectionUtil.getNMSClass("EntityPlayer")));
MH_HEALTH_UPDATE = lookup.findVirtual(ReflectionUtil.getBukkitClass("entity.CraftPlayer"), "triggerHealthUpdate", MethodType.methodType(Void.TYPE));
```
I think it then would be an issue for paper specifically? 
try with spigot, and see if you can reproduce.

**Comment by s1h4d0w** [comment]
Sorry for not getting back to you, but I've tested with other plugins and the same issue happened. I've resorted to disabling the `/skin` command in worldguard regions which has for now stopped people abusing it. They might be able to still do it if they do it near a region border, but if that is the case I'll just disable the plugin altogether. 

I reported the issue to Paper but they won't even look at it because I'm running a 1.12 server (anarchy, so need the performance). I've been dealing with people trying to find exploits a lot recently while I already have very little time, so I've just given up on this. If they manage to still abuse the bug I'll simply disable the plugins and they can play with whatever skin they join with.
